File: Find Common Characters.cpp
class Solution {
public:
    vector<int> count(string word) {
        vector<int> counter(26, 0);
        for (char&c : word) {
            counter[(int) c - 97]++;
        }
        return counter;
    }
    vector<string> commonChars(vector<string>& words) {
        vector<int> counter = count(words[0]);
        int n = words.size();

        for (int i = 1; i < n; i++) {
            vector<int> counter2 = count(words[i]);
            for (int j = 0; j < 26; j++) {
                counter[j] = min(counter[j], counter2[j]);
            }
        }

        vector<string> ans;
        for (int i = 0; i < 26; i++) {
            for (int j = 0; j < counter[i]; j++) {
                string str(1, (char) (i+97));
                ans.push_back(str);
            }
        }
        return ans;
    }
};

File: Find Common Characters.py
class Solution:
    def commonChars(self, words: List[str]) -> List[str]:
        def count(word):
            counter = [0] * 26
            for c in word:
                counter[ord(c) - 97] += 1
            return counter

        counter = count(words[0])
        n = len(words)
        for i in range(1, n):
            counter2 = count(words[i])
            for j in range(26):
                counter[j] = min(counter[j], counter2[j])
        ans = []
        for i, c in enumerate(counter):
            for _ in range(c):
                ans.append(chr(i + 97))
        return ans
